content/doc/developer/index.html.haml

```diff
@@ -0,0 +1,39 @@
+---
+title: "Jenkins Developer Guide"
+layout: developer
+---
+
+The Jenkins Developer Guide serves as a reference for core and plugin developers.
+
+%ol
+  - site.devbook.chapters.each do |chapter|
+    %li{:style => 'font-size: 24px;'}
+      %a{:href => chapter.page.url}
+        = chapter.title
+      - if chapter.page.wip
+        - if !chapter.page.references || chapter.page.references.length == 0
+          %i.icon-warning{:title => 'Early work in progress'}
+            &nbsp;
+        - else
+          %i.icon-warning{:title => 'Partial work in progress', :style => 'opacity: 0.5;'}
+            &nbsp;
+      - if chapter.page.summary
+        %p{:style => 'font-size: 14px;'}
+          = chapter.page.summary
+      %ol
+        - chapter.sections.each do | section |
+          %li{:style => 'font-size: 18px;'}
+            %a{:href => section.page.url}
+
+              = section.title
+
+            - if section.page.wip
+              - if !section.page.references || section.page.references.length == 0
+                %i.icon-warning{:title => 'Work in progress'}
+                  &nbsp;
+              - else
+                %i.icon-warning{:title => 'Partial work in progress', :style => 'opacity: 0.5;'}
+                  &nbsp;
+  %li{:style => 'font-size: 24px;'}
+    %a{:href => 'guides'}
+      Appendix C: Index of All How-To Guides
```

Comment:
I haven't figured out how to make the site assign the sequence, so I'm probably just off by one.